src/redis-cli.c

```diff
@@ -1266,6 +1267,10 @@ static int cliReadReply(int output_raw_strings) {
                 slot, config.hostip, config.hostport);
         config.cluster_reissue_command = 1;
         cliRefreshPrompt();
+    } else if (config.set_errcode && reply->type == REDIS_REPLY_ERROR) {
+        fprintf(stderr,"%s\n",reply->str);
+        exit(1);
```

Comment:
@hwware In addition to aborting with an error code, this also introduces other changes as the error is now sent to `stderr` rather than `stdout` and it does not conform to the `cliFormatReply` formatting. Is that intentional? We should consider maintaining compatibility here.

Reply:
hi @yossigo , thanks for your comments, yes this is intentional, since when cli abort with error code, the stderr will always been set based on other parts of code, for example when context error happens and we have to exit with an error: https://github.com/redis/redis/blob/unstable/src/redis-cli.c#L873 . the cliFormatReply based on my understanding is formatting reply for "no error", even though this function contains a error case we still don't consider the execution fails and the exit code is still 0.